src/Server.php

```diff
@@ -75,26 +77,41 @@
  */
 class Server extends EventEmitter
 {
-    private $callback;
+    /**
+     * @var MiddlewareStackInterface
+     */
+    private $middlewareStack;
 
     /**
-     * Creates an HTTP server that invokes the given callback for each incoming HTTP request
+     * Creates an HTTP server that invokes the given middleware stack for each
+     * incoming HTTP request. The middleware stack is either a concrete class
+     * implementing `React\Http\MiddlewareStackInterface` or an array of
+     * `React\Http\MiddlewareInterface`s.
      *
      * In order to process any connections, the server needs to be attached to an
      * instance of `React\Socket\ServerInterface` which emits underlying streaming
      * connections in order to then parse incoming data as HTTP.
      * See also [listen()](#listen) for more details.
      *
-     * @param callable $callback
+     * @param MiddlewareInterface[]|MiddlewareStackInterface $middlewares
      * @see self::listen()
      */
-    public function __construct($callback)
+    public function __construct($middlewares)
     {
-        if (!is_callable($callback)) {
-            throw new \InvalidArgumentException();
+        if (is_array($middlewares)) {
+            $this->middlewareStack = new MiddlewareStack(
+                new Response(404),
+                $middlewares
+            );
+            return;
+        }
+
+        if ($middlewares instanceof MiddlewareStackInterface) {
+            $this->middlewareStack = $middlewares;
+            return;
         }
 
-        $this->callback = $callback;
+        throw new \InvalidArgumentException();
```

Comment:
Exception message missing explaining what types of arguments can be given to the constructor.